web/docs/language/features.md

```diff
@@ -443,6 +443,149 @@ import { isPrismaError, prismaErrorToHttpError } from '@wasp/utils.js'
+
+#### pg-boss
+
+We have selected [pg-boss](https://github.com/timgit/pg-boss/) as our first job executor to handle the low-volume, basic job queue workloads many web applications have (think sending emails, connecting to external HTTP APIs, etc.). By using PostgreSQL (and [SKIP LOCKED](https://www.2ndquadrant.com/en/blog/what-is-select-skip-locked-for-in-postgresql-9-5/)) as it's storage and synchronization mechanism, it allows us to provide many job queue pros without any additional infrastructure or complex management.
+
+### Basic job definition and usage
+
+To register a `job` in Wasp, simply add a declaration with a reference to an `async` function, like the following:
+
+```css title="main.wasp"
+job mySpecialJob {
+  executor: PgBoss,
+  perform: {
+    fn: import { foo } from "@ext/jobs/bar.js",
+    options: {=json { "retryLimit": 1 } json=} // optional
+  }
+}
+```
+
+Then, in your [Operations](/docs/language/features#queries-and-actions-aka-operations) or [setupFn](/docs/language/features#setupfn-extimport-optional), you can enqueue work to be done:
+```js
+import { mySpecialJob } from '@wasp/jobs/mySpecialJob.js'
+
+const submittedJob = await mySpecialJob.submit({ job: "args" })
+console.log(await submittedJob.pgBoss.details())
+
+// Or, if you'd prefer it to execute in the future, just add a .delay().
+// It takes a number of seconds, Date, or ISO date string.
+await mySpecialJob.delay(10).submit({ job: "args" })
+```
+
+And that is it! Your job will be executed by the job executor as if you called `foo({ job: "args" })`.
+
+### Recurring jobs
+
+If you have work that needs to be done on some recurring basis, you can add a `schedule` to your job declaration:
+
+```css  {6-10} title="main.wasp"
+job mySpecialJob {
+  executor: PgBoss,
+  perform: {
+    fn: import { foo } from "@ext/jobs/bar.js"
+  },
+  schedule: {
+    cron: "0 * * * *",
+    performFnArg: {=json { "job": "args" } json=}, // optional
+    options: {=json { "retryLimit": 2 } json=} // optional
+  }
+}
+```
+
+In this example, you do _not_ need to invoke anything in JavaScript. You can imagine `foo({ "job": "args" })` getting automatically scheduled and invoked for you every hour.
+
+### Fields
+
+#### `executor: JobExecutor` (required)
+The executor can be either `Passthrough` or `PgBoss`. `Passthrough` is a zero-dependency executor used mainly for testing purposes. `PgBoss` is the recommended executor for low-volume, production use cases, and requires your `app.db.system` to be `PostgreSQL`.
+
+####  `perform: dict` (required)
+
+  - ##### `fn: fn` (required)
+  An `async` JavaScript function of work to be performed. It can take any number of arguments.
+  
+  - ##### `options: JSON` (optional)
```

Comment:
Ok I know it is a bit late to ask these questions :D, but what if we would name these `executorOptions`? If I got it right, at this point we have 3 different `options` fields, and I am getting a bit confused as to which options are what.

Reply:
Yeah, I think renaming to `executorOptions` is fine. In this case, do we still want to namespace them under the executor name, or is that now explicit enough?

Reply:
I was reffering to top level options, which are I believe different from `perform` options, right?
Because these top level options are setting up stuff for the executor, while `perform.options` are setting stuff up for the job?

Hm so it would be:
```
job MyJob {
  executor: PgBoss,
  executorOptions: { ... },
  perform: {
    fn: ...,
    options: {
      pgBoss: ...
    }
  },
  schedule: {
    ...,
    options: {
      pgBoss: ...
    }
  }
}
```

Hm ok. So `pgBoss` in `perform.options` is useful because it makes it clear that these are `pgBoss` related options, not general "perform/job" options.
Same for `schedule`.
As for `executorOptions` - well you are right, maybe we should also have `pgBoss` there. Since maybe they can provide general executor options also.

Reply:
As of now, there is not notion of options at the top level (alongside `executor`) just those scoped under `perform` and `schedule`. However, that could be a good way to put it if we need them up top.